clang/lib/Driver/ToolChains/SYCL.h

```diff
@@ -16,6 +16,159 @@
 namespace clang {
 namespace driver {
 
+// List of architectures (Intel CPU, Intel GPU, AMD GPU, NVidia GPU)
+// that support SYCL offloading.
+enum class SYCLSupportedOffloadArchs {
+  // Intel CPUs
+  UNKNOWN,
+  SKYLAKEAVX512,
+  COREAVX2,
+  COREI7AVX,
+  COREI7,
+  WESTMERE,
+  SANDYBRIDGE,
+  IVYBRIDGE,
+  BROADWELL,
+  COFFEELAKE,
+  ALDERLAKE,
+  SKYLAKE,
+  SKX,
+  CASCADELAKE,
+  ICELAKECLIENT,
+  ICELAKESERVER,
+  SAPPHIRERAPIDS,
+  GRANITERAPIDS,
+  // Intel GPUs
+  BDW,
+  SKL,
+  KBL,
+  CFL,
+  APL,
+  BXT,
+  GLK,
+  WHL,
+  AML,
+  CML,
+  ICLLP,
+  ICL,
+  EHL,
+  JSL,
+  TGLLP,
+  TGL,
+  RKL,
+  ADL_S,
+  RPL_S,
+  ADL_P,
+  ADL_N,
+  DG1,
+  ACM_G10,
+  DG2_G10,
+  ACM_G11,
+  DG2_G11,
+  ACM_G12,
+  DG2_G12,
+  PVC,
+  PVC_VG,
+  MTL_U,
+  MTL_S,
+  ARL_U,
+  ARL_S,
+  MTL_H,
+  ARL_H,
+  BMG_G21,
+  LNL_M,
+  // AMD GPUs
+  GFX700,
+  GFX701,
+  GFX702,
+  GFX801,
+  GFX802,
+  GFX803,
+  GFX805,
+  GFX810,
+  GFX900,
+  GFX902,
+  GFX904,
+  GFX906,
+  GFX908,
+  GFX909,
+  GFX90A,
+  GFX90C,
+  GFX940,
+  GFX941,
+  GFX942,
+  GFX1010,
+  GFX1011,
+  GFX1012,
+  GFX1013,
+  GFX1030,
+  GFX1031,
+  GFX1032,
+  GFX1033,
+  GFX1034,
+  GFX1035,
+  GFX1036,
+  GFX1100,
+  GFX1101,
+  GFX1102,
+  GFX1103,
+  GFX1150,
+  GFX1151,
+  GFX1200,
+  GFX1201,
+  // NVidia GPUs.
+  SM_50,
+  SM_52,
+  SM_53,
+  SM_60,
+  SM_61,
+  SM_62,
+  SM_70,
+  SM_72,
+  SM_75,
+  SM_80,
+  SM_86,
+  SM_87,
+  SM_89,
+  SM_90,
+  SM_90A
+};
+
+// Check if the given Arch value is a valid SYCL supported AMD GPU.
+static inline bool IsSYCLSupportedAMDGPUArch(SYCLSupportedOffloadArchs Arch) {
+  return Arch >= SYCLSupportedOffloadArchs::GFX700 &&
+         Arch <= SYCLSupportedOffloadArchs::GFX1201;
+}
+
+// Check if the given Arch value is a valid SYCL supported NVidia GPU.
+static inline bool
+IsSYCLSupportedNVidiaGPUArch(SYCLSupportedOffloadArchs Arch) {
+  return Arch >= SYCLSupportedOffloadArchs::SM_50 &&
+         Arch <= SYCLSupportedOffloadArchs::SM_90A;
+}
+
+// Check if the given Arch value is a valid SYCL supported Intel CPU.
+static inline bool IsSYCLSupportedIntelCPUArch(SYCLSupportedOffloadArchs Arch) {
+  return Arch >= SYCLSupportedOffloadArchs::SKYLAKEAVX512 &&
+         Arch < SYCLSupportedOffloadArchs::BDW;
```

Comment:
For consistency with the others:
```suggestion
         Arch <= SYCLSupportedOffloadArchs::GRANITERAPIDS;
```